Spec/Auth.swift

```diff
@@ -626,16 +627,24 @@ class Auth : QuickSpec {
                     authOptions.authMethod = "POST"
                     let tokenParams = ARTTokenParams()
                     tokenParams.ttl = 30.0
+                    tokenParams.clientId = "anId"
                     
                     // AuthOptions
-                    let mergedOptions = rest.auth.mergeOptions(authOptions)
-                    expect(mergedOptions.authUrl) == NSURL(string: "http://test.ably.io")
-                    expect(mergedOptions.authMethod) == "POST"
+                    let replacedOptions = rest.auth.replaceOptions(authOptions)
+                    expect(replacedOptions.authUrl) == NSURL(string: "http://test.ably.io")
+                    expect(replacedOptions.authMethod) == "POST"
+                    expect(replacedOptions.key).to(beNil())
                     // TokenParams
                     let mergedParams = rest.auth.mergeParams(tokenParams)
                     expect(mergedParams.ttl) == 30.0
+                    
+                    let tokenParams2 = ARTTokenParams()
+                    tokenParams2.ttl = 25.0
+                    
+                    let mergedParams2 = rest.auth.mergeParams(tokenParams2)
+                    expect(mergedParams2.ttl) == 25.0
+                    expect(mergedParams2.clientId).to(beNil())
```

Comment:
I am not really following the test code here.  Thee test code is not testing the `requestToken` method here but instead you are testing private undocumented APIs.  The point of the test is to "should not merge with the configured params and options but instead replace all corresponding values, even when @null@" yet this is not asserting that?